src/udiskscrypttabmonitor.c

```diff
@@ -346,6 +346,51 @@ have_entry (UDisksCrypttabMonitor *monitor,
   return ret;
 }
 
+/**
+ * split_crypttab_line:
+ *
+ * Splits line from /etc/crypttab to parts (device, name, password and options).
+ * This function also handles columns with multiple deliminters (tabs or spaces).
+ *
+ */
+static gchar **
+split_crypttab_line (const gchar *line, guint *num_tokens)
+{
+  GSList *list = NULL;
+  gint next = 0;
+  gchar **tokens = NULL;
+  gchar *line_copy = NULL;
+  gchar *str1, *str2, *token, *subtoken;
+  gchar *saveptr1, *saveptr2;
+
+  line_copy = g_strdup (line);  // strtok will modify this
+
+  for (str1 = line_copy; ; str1 = NULL)
+    {
+      token = strtok_r (str1, " ", &saveptr1);
+      if (token == NULL)
+          break;
+      for (str2 = token; ; str2 = NULL)
+        {
+            subtoken = strtok_r (str2, "\t", &saveptr2);
+            if (subtoken == NULL)
+                break;
+            list = g_slist_append (list, g_strdup (subtoken));
+        }
+     }
+
+  *num_tokens = g_slist_length (list);
+  tokens = g_new0 (gchar *, *num_tokens + 1);
+  for (GSList *list_p = list; list_p != NULL; list_p = list_p->next)
```

Comment:
Just a hint (no need to change this): you may use `GPtrArray` instead of `GSList` and call `g_ptr_array_free()` that will return you `gpointer *` that you can just retype to `gchar **`. An easy way how to dynamically build an array of strings...